hypervisor/src/mshv/mod.rs

```diff
@@ -1573,17 +1506,56 @@ impl MshvVcpu {
         ghcb_gpa: u64,
     ) -> std::result::Result<cpu::VmExit, cpu::HypervisorCpuError> {
         // Clear the SW_EXIT_INFO1 register to indicate no error
-        let mut swei1_rw_gpa_arg = mshv_bindings::mshv_read_write_gpa {
-            base_gpa: ghcb_gpa + GHCB_SW_EXITINFO1_OFFSET,
-            byte_count: std::mem::size_of::<u64>() as u32,
-            ..Default::default()
-        };
-        self.fd
-            .gpa_write(&mut swei1_rw_gpa_arg)
-            .map_err(|e| cpu::HypervisorCpuError::GpaWrite(e.into()))?;
+        self.gpa_write(ghcb_gpa + GHCB_SW_EXITINFO1_OFFSET, &[0; 4])?;
 
         Ok(cpu::VmExit::Ignore)
     }
+
+    #[cfg(feature = "sev_snp")]
+    fn gpa_read(
+        &self,
+        gpa: u64,
+        data: &mut [u8],
+    ) -> std::result::Result<(), cpu::HypervisorCpuError> {
+        const CHUNK_SIZE: usize = 16;
```

Comment:
Can you please use the constant from mshv-bindings instead of redefining it here: `mshv_bindings::HV_READ_WRITE_GPA_MAX_SIZE`?